src/mon/Monitor.cc

```diff
@@ -5364,6 +5385,8 @@ void Monitor::handle_subscribe(MonOpRequestRef op)
     }
     else if (p->first == "NVMeofGw") {
         nvmegwmon()->check_sub(s->sub_map[p->first]);
+    } else if (p->first == "quorum_change") {
+      send_quorum_changed(s->sub_map[p->first]);
```

Comment:
I'm puzzled: why not just subscribe to the monmap with the aux connection to the monitor?

Reply:
@batrick i'm trying for now to keep it minimal change, for the POC, but yes, we can also get the monmap from all the monitors that in aux list

Reply:
@batrick so, you mean to use:
 else if (p->first == "monmap") {
      monmon()->check_sub(s->sub_map[p->first]);
  
  and send the quorum list whenever we have monmap, or check for monmap?
  it sounds reasonable, we will then get the monmap from all the aux list and the quorum list, we won't do anything yet with the others monmap accept the active connection one.

Reply:
> and send the quorum list whenever we have monmap, or check for monmap?

If we're trying to make the client resilient to quorum changes, I think subscribing to the monmap from a majority of monitors (i.e. a valid quorum) makes sense. Then pick one of those monitors (perhaps the closest one when using stretch clusters) for the other subs / operations.